plugins/woocommerce-blocks/assets/js/blocks/product-collection/deprecated.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+import metadata from './block.json';
+import save from './save';
+import type { ProductCollectionQuery } from './types';
+
+const v1 = {
+	attributes: {
+		...metadata.attributes,
+		query: {
+			type: 'object',
+			properties: {
+				woocommerceOnSale: {
+					type: 'boolean',
+					default: false,
+				},
+			},
+		},
+	},
```

Comment:
Out of curiosity, Do we even need to add `properties` inside `query`? Won't it work if we just remove it? i.e.

```suggestion
	attributes: {
		...metadata.attributes,
	},
```

Reply:
I'll work too, but I wanted to be specific about which property was deprecated and what was its previous definition.

Reply:
Got it, thanks 🚀